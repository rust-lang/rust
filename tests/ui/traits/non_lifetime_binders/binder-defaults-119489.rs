#![feature(non_lifetime_binders, generic_const_exprs)]
//~^ WARN the feature `non_lifetime_binders` is incomplete
//~| WARN the feature `generic_const_exprs` is incomplete

fn fun()
where
    for<T = (), const N: usize = 1> ():,
    //~^ ERROR late-bound const parameters cannot be used currently
    //~| ERROR defaults for generic parameters are not allowed in `for<...>` binders
    //~| ERROR defaults for generic parameters are not allowed in `for<...>` binders
{}

fn main() {}
