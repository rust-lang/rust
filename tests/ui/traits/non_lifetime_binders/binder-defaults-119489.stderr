error: late-bound const parameters cannot be used currently
  --> $DIR/binder-defaults-119489.rs:7:23
   |
LL |     for<T = (), const N: usize = 1> ():,
   |                       ^

warning: the feature `non_lifetime_binders` is incomplete and may not be safe to use and/or cause compiler crashes
  --> $DIR/binder-defaults-119489.rs:1:12
   |
LL | #![feature(non_lifetime_binders, generic_const_exprs)]
   |            ^^^^^^^^^^^^^^^^^^^^
   |
   = note: see issue #108185 <https://github.com/rust-lang/rust/issues/108185> for more information
   = note: `#[warn(incomplete_features)]` on by default

warning: the feature `generic_const_exprs` is incomplete and may not be safe to use and/or cause compiler crashes
  --> $DIR/binder-defaults-119489.rs:1:34
   |
LL | #![feature(non_lifetime_binders, generic_const_exprs)]
   |                                  ^^^^^^^^^^^^^^^^^^^
   |
   = note: see issue #76560 <https://github.com/rust-lang/rust/issues/76560> for more information

error: defaults for generic parameters are not allowed in `for<...>` binders
  --> $DIR/binder-defaults-119489.rs:7:9
   |
LL |     for<T = (), const N: usize = 1> ():,
   |         ^^^^^^

error: defaults for generic parameters are not allowed in `for<...>` binders
  --> $DIR/binder-defaults-119489.rs:7:17
   |
LL |     for<T = (), const N: usize = 1> ():,
   |                 ^^^^^^^^^^^^^^^^^^

error: aborting due to 3 previous errors; 2 warnings emitted

